src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheckTest.java

```diff
@@ -93,6 +102,32 @@ public void testIllegalClasses()
     }
 
     @Test
+    public void testIllegalClassesCleared()
+            throws Exception {
+        final File input = new File(getPath("InputIllegalImportTestExcludes.java"));
+        final DetailAST firstImport = TestUtils.parseFile(input).getNextSibling();
+        final DetailAST secondImport = firstImport.getNextSibling();
+        final IllegalImportCheck check = new IllegalImportCheck();
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(IllegalImportCheck.class);
+        check.configure(checkConfig);
+        check.setFileContents(new FileContents(new FileText(input, "UTF-8")));
+        check.setRegexp(true);
+
+        check.setIllegalClasses("java.sql.Connection");
+        check.setIllegalClasses("java.nio.file.Files");
+        check.visitToken(firstImport);
+        check.visitToken(secondImport);
```

Comment:
I need good reason why this can not be tested by `verify`.
All should be reproducible by java files, if not possible - it is dead and useless code .
if still good to keep - test method should explain usage of hacked test methodology

Reply:
This method is used only in tests, so it may be reasonable to delete it, do you agree?

Reply:
Check property- http://checkstyle.sourceforge.net/config_imports.html#IllegalImport `illegalClasses`